Kudu.Contracts/SiteExtensions/ISiteExtensionManager.cs

```diff
@@ -1,20 +1,20 @@
-﻿using System;
-using System.Collections.Generic;
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
 
 namespace Kudu.Contracts.SiteExtensions
 {
     public interface ISiteExtensionManager
     {
-        IEnumerable<SiteExtensionInfo> GetRemoteExtensions(string filter, bool allowPrereleaseVersions, string feedUrl);
+        Task<IEnumerable<SiteExtensionInfo>> GetRemoteExtensions(string filter, bool allowPrereleaseVersions, string feedUrl);
 
-        SiteExtensionInfo GetRemoteExtension(string id, string version, string feedUrl);
+        Task<SiteExtensionInfo> GetRemoteExtension(string id, string version, string feedUrl);
 
-        IEnumerable<SiteExtensionInfo> GetLocalExtensions(string filter, bool checkLatest);
+        Task<IEnumerable<SiteExtensionInfo>> GetLocalExtensions(string filter, bool checkLatest);
 
-        SiteExtensionInfo GetLocalExtension(string id, bool checkLatest);
+        Task<SiteExtensionInfo> GetLocalExtension(string id, bool checkLatest);
 
-        SiteExtensionInfo InstallExtension(string id, string version, string feedUrl);
+        Task<SiteExtensionInfo> InstallExtension(string id, string version, string feedUrl);
```

Comment:
@davidebbo Sooner than later, we need to make Install (and perhaps Uninstall) extension a true CSM async operation pattern.